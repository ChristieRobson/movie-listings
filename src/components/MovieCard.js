import React from 'react';
import { Card, CardImg, CardText, CardBody, CardTitle, CardSubtitle } from 'reactstrap';

export const MovieCard = (props) => {
    return (
        <div className="col-12 col-md-6 col-lg-4">
            <Card>
                <CardImg
                    top
                    src={`https://image.tmdb.org/t/p/original/${props.poster_path}`}
                    alt="Card image cap" />
                <CardBody>
                    <CardTitle>{props.title}</CardTitle>
                    <CardSubtitle>Popularity: {props.popularity}</CardSubtitle>
                    <CardText>
                        Genres: {props.genres && props.genres.map(
                            (genre) => {
                                return (<span key={genre.id}>{genre.name} </span>)
                            }
                        )}
                    </CardText>
                </CardBody>
            </Card>
        </div>
    )
};

//
// "vote_count": 213,
//     "id": 460668,
//     "video": false,
//     "vote_average": 6.3,
//     "title": "I Feel Pretty",
//     "popularity": 41.362,
//     "poster_path": "\/iuPs45XIxfARKPLEkCGXWUrBrTR.jpg",
//     "original_language": "en",
//     "original_title": "I Feel Pretty",
//     "genre_ids": [
//     35
// ],
//     "backdrop_path": "\/3Zb4Du20mvKl0jlOtQ6lCFhpiiT.jpg",
//     "adult": false,
//     "overview": "A head injury causes a woman to develop an extraordinary amount of confidence and believe she's drop dead gorgeous.",
//     // "release_date": "2018-04-19"