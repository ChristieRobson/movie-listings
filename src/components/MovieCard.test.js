import React from 'react';
import { shallow } from 'enzyme';
import { MovieCard } from './MovieCard';

it('renders MovieCard', () => {
    const movieStub = {
        poster_path: 'abc123',
        title: 'A movie title',
        popularity: 88.123,
        genres : [
            {
                "id": 28,
                "name": "Action"
            },
            {
                "id": 12,
                "name": "Adventure"
            }
        ]
    };
    let component = shallow(<MovieCard {...movieStub}/>);
    expect(component).toMatchSnapshot();
});