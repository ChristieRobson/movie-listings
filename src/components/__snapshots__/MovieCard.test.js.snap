// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MovieCard 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    genres={
        Array [
            Object {
              "id": 28,
              "name": "Action",
            },
            Object {
              "id": 12,
              "name": "Adventure",
            },
          ]
    }
    popularity={88.123}
    poster_path="abc123"
    title="A movie title"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Card
        tag="div"
>
        <CardImg
                alt="Poster for A movie title"
                src="https://image.tmdb.org/t/p/original/abc123"
                tag="img"
                top={true}
        />
        <CardBody
                tag="div"
        >
                <CardTitle
                        tag="h5"
                >
                        A movie title
                </CardTitle>
                <CardSubtitle
                        tag="h6"
                >
                        Popularity: 
                        88.123
                </CardSubtitle>
                <CardText
                        tag="p"
                >
                        Genres: 
                        <span>
                                Action
                                 
                        </span>
                        <span>
                                Adventure
                                 
                        </span>
                </CardText>
        </CardBody>
</Card>,
      "className": "col-12 col-md-6 col-lg-4",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <CardImg
            alt="Poster for A movie title"
            src="https://image.tmdb.org/t/p/original/abc123"
            tag="img"
            top={true}
/>,
          <CardBody
            tag="div"
>
            <CardTitle
                        tag="h5"
            >
                        A movie title
            </CardTitle>
            <CardSubtitle
                        tag="h6"
            >
                        Popularity: 
                        88.123
            </CardSubtitle>
            <CardText
                        tag="p"
            >
                        Genres: 
                        <span>
                                    Action
                                     
                        </span>
                        <span>
                                    Adventure
                                     
                        </span>
            </CardText>
</CardBody>,
        ],
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "alt": "Poster for A movie title",
            "src": "https://image.tmdb.org/t/p/original/abc123",
            "tag": "img",
            "top": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <CardTitle
                tag="h5"
>
                A movie title
</CardTitle>,
              <CardSubtitle
                tag="h6"
>
                Popularity: 
                88.123
</CardSubtitle>,
              <CardText
                tag="p"
>
                Genres: 
                <span>
                                Action
                                 
                </span>
                <span>
                                Adventure
                                 
                </span>
</CardText>,
            ],
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "A movie title",
                "tag": "h5",
              },
              "ref": null,
              "rendered": "A movie title",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  "Popularity: ",
                  88.123,
                ],
                "tag": "h6",
              },
              "ref": null,
              "rendered": Array [
                "Popularity: ",
                88.123,
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  "Genres: ",
                  Array [
                    <span>
                      Action
                       
</span>,
                    <span>
                      Adventure
                       
</span>,
                  ],
                ],
                "tag": "p",
              },
              "ref": null,
              "rendered": Array [
                "Genres: ",
                Object {
                  "instance": null,
                  "key": "28",
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Action",
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Action",
                    " ",
                  ],
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": "12",
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Adventure",
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Adventure",
                    " ",
                  ],
                  "type": "span",
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Card
          tag="div"
>
          <CardImg
                    alt="Poster for A movie title"
                    src="https://image.tmdb.org/t/p/original/abc123"
                    tag="img"
                    top={true}
          />
          <CardBody
                    tag="div"
          >
                    <CardTitle
                              tag="h5"
                    >
                              A movie title
                    </CardTitle>
                    <CardSubtitle
                              tag="h6"
                    >
                              Popularity: 
                              88.123
                    </CardSubtitle>
                    <CardText
                              tag="p"
                    >
                              Genres: 
                              <span>
                                        Action
                                         
                              </span>
                              <span>
                                        Adventure
                                         
                              </span>
                    </CardText>
          </CardBody>
</Card>,
        "className": "col-12 col-md-6 col-lg-4",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <CardImg
              alt="Poster for A movie title"
              src="https://image.tmdb.org/t/p/original/abc123"
              tag="img"
              top={true}
/>,
            <CardBody
              tag="div"
>
              <CardTitle
                            tag="h5"
              >
                            A movie title
              </CardTitle>
              <CardSubtitle
                            tag="h6"
              >
                            Popularity: 
                            88.123
              </CardSubtitle>
              <CardText
                            tag="p"
              >
                            Genres: 
                            <span>
                                          Action
                                           
                            </span>
                            <span>
                                          Adventure
                                           
                            </span>
              </CardText>
</CardBody>,
          ],
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "alt": "Poster for A movie title",
              "src": "https://image.tmdb.org/t/p/original/abc123",
              "tag": "img",
              "top": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <CardTitle
                  tag="h5"
>
                  A movie title
</CardTitle>,
                <CardSubtitle
                  tag="h6"
>
                  Popularity: 
                  88.123
</CardSubtitle>,
                <CardText
                  tag="p"
>
                  Genres: 
                  <span>
                                    Action
                                     
                  </span>
                  <span>
                                    Adventure
                                     
                  </span>
</CardText>,
              ],
              "tag": "div",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "A movie title",
                  "tag": "h5",
                },
                "ref": null,
                "rendered": "A movie title",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    "Popularity: ",
                    88.123,
                  ],
                  "tag": "h6",
                },
                "ref": null,
                "rendered": Array [
                  "Popularity: ",
                  88.123,
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    "Genres: ",
                    Array [
                      <span>
                        Action
                         
</span>,
                      <span>
                        Adventure
                         
</span>,
                    ],
                  ],
                  "tag": "p",
                },
                "ref": null,
                "rendered": Array [
                  "Genres: ",
                  Object {
                    "instance": null,
                    "key": "28",
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Action",
                        " ",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Action",
                      " ",
                    ],
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": "12",
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Adventure",
                        " ",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Adventure",
                      " ",
                    ],
                    "type": "span",
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
